### PR TITLE
overide env.getNpmPaths

### PR DESCRIPTION
**Pull request by tomer-epstein**

### CODE REVIEW

File: backend/src/yeomanui.ts [hunk 1]
```diff
@@ -13,6 +13,9 @@ import { IRpc } from "@sap-devx/webview-rpc/out.ext/rpc-common";
 import Generator = require("yeoman-generator");
 import { GeneratorType, GeneratorFilter } from "./filter";
 
+const uniq = require('array-uniq');
```
Comment (slavik-lvovsky) [inline backend/src/yeomanui.ts]: you should use _.uniq() from lodash instead of using this package

File: backend/src/yeomanui.ts [hunk 2]
```diff
@@ -73,7 +76,28 @@ export class YeomanUI {
     // on the other hand, we never look for newly installed generators...
 
     const promise: Promise<IPrompt> = new Promise(resolve => {
+      const cwd = path.join(os.homedir(), "projects");
       const env: Environment.Options = Environment.createEnv();
+      const envGetNpmPaths: Function = env.getNpmPaths;
```
Comment (slavik-lvovsky) [inline backend/src/yeomanui.ts]: lint error:
ERROR: src/yeomanui.ts:81:29 - Don't use 'Function' as a type. Avoid using the `Function` type. Prefer a specific function type, like `() => void`.

File: backend/src/yeomanui.ts [hunk 3]
```diff
@@ -73,7 +76,28 @@ export class YeomanUI {
...
+      env.getNpmPaths = function (localOnly:any = false) {
```
Comment (slavik-lvovsky) [inline backend/src/yeomanui.ts]: need to add test that checks this overwritten function

File: backend/src/yeomanui.ts [hunk 4]
```diff
@@ -13,6 +13,9 @@ import { IRpc } from "@sap-devx/webview-rpc/out.ext/rpc-common";
 import Generator = require("yeoman-generator");
 import { GeneratorType, GeneratorFilter } from "./filter";
 
+const uniq = require('array-uniq');
+const win32 = process.platform === 'win32';
```
Comment (slavik-lvovsky) [inline backend/src/yeomanui.ts]: it is a boolean, but from the name it is very hard to know.
Change it to something like: isWin32

File: backend/src/yeomanui.ts [hunk 5]
```diff
@@ -73,7 +76,28 @@ export class YeomanUI {
     // on the other hand, we never look for newly installed generators...
 
     const promise: Promise<IPrompt> = new Promise(resolve => {
+      const cwd = path.join(os.homedir(), "projects");
```
Comment (idoprz) [inline backend/src/yeomanui.ts]: This should be configurable by the user (as a vscode settings)

Comment (tomer-epstein) [reply backend/src/yeomanui.ts]: Issue created - https://github.com/SAP/yeoman-ui/issues/57

File: backend/src/yeomanui.ts [hunk 6]
```diff
@@ -73,7 +75,28 @@ export class YeomanUI {
...
       const env: Environment.Options = Environment.createEnv();
+      const envGetNpmPaths: () => any = env.getNpmPaths;
+      env.getNpmPaths = function (localOnly:any = false) {
+        // Start with the local paths derived by cwd in vscode 
+        // (as opposed to cwd of the plugin host process which is what is used by yeoman/environment)
+        const localPaths: any[] = [];
+    
+        // Walk up the CWD and add `node_modules/` folder lookup on each level
```
Comment (slavik-lvovsky) [inline backend/src/yeomanui.ts]: Need to create a new method called getEnvironment that will return an environment and will contain all the new getNpmPath logic

File: backend/src/yeomanui.ts [hunk 7]
```diff
@@ -13,6 +13,8 @@ import { IRpc } from "@sap-devx/webview-rpc/out.ext/rpc-common";
 import Generator = require("yeoman-generator");
 import { GeneratorType, GeneratorFilter } from "./filter";
 
+const isWin32 = process.platform === 'win32';
+
```
Comment (tomer-epstein) [inline backend/src/yeomanui.ts]: Remove. (duplicate)

**Pull request merged**